src/Microsoft.DotNet.Build.CloudTestTasks/UploadClient.cs

```diff
@@ -208,6 +210,57 @@ public string EncodeBlockIds(int numberOfBlocks, int lengthOfId)
                 }
             }
         }
+
+        public async Task<bool> FileEqualsExistingBlobAsync(
+            string accountName,
+            string accountKey,
+            string containerName,
+            string filePath,
+            string destinationBlob,
+            int uploadTimeout)
+        {
+            using (var client = new HttpClient
```

Comment:
I'm sure we already have existing code that does similar stuff, we should make sure we re-using it instead of making another copy of it.

Reply:
I'll take another look. I didn't see anything first time around--it looked like the `AzureHelper` methods were the highest-level we had. I did copy those lines from somewhere, so I could just go ahead and centralize it now.

Reply:
Here's an attempt to do some refactoring to use more centralized helpers: https://github.com/dagood/buildtools/commit/a52905777c02ff50ebbf01e53501375895dd24a4. Didn't include it in this PR because I'm not sure it's worth testing/reviewing in the context of this (time-sensitive) change. Also, on the backlog we plan to remove the custom Azure code anyway.

Reply:
I'm OK not including it yet but I want to make sure we have an issue tracking it.

Reply:
Thanks, created https://github.com/dotnet/buildtools/issues/1897.